fronted/src/components/App.js

```diff
@@ -152,34 +177,35 @@ function App() {
     setIsLoggedIn(false);
     localStorage.removeItem('token');
     localStorage.removeItem('latest-search');
-    setCardsToRender([]);
+    setCardsToRender(false);
     setSavedArticles([]);
-    closePopups();;
+    setKeyword('');
+    closePopups();
   }
 
```

Comment:
You can add an `Escape` listener like this:

```jsx
 useEffect(() => {
      const closeByEscape = (e) => {
        if (e.key === 'Escape') {
          closeAllPopups();
        }
      }

      document.addEventListener('keydown', closeByEscape)
      
      return () => document.removeEventListener('keydown', closeByEscape)
  }, [])

```

Here you define function `closeByEscape` inside the `useEffect` so that it could maintain a fixed reference if the component rerenders.
And of course we need to return a `clean up` function to `remove` the listener if the component unmounts.
And don’t forget about an `empty dependencies array` in order to add the listener only once.